Services/WebServices/ECS/classes/Consent/class.ilECSUserConsentModalGUI.php

```diff
@@ -0,0 +1,348 @@
+        $title->setValue(
+            ilObject::_lookupTitle(ilObject::_lookupObjId($this->ref_id))
+        );
+        $form->addItem($title);
+
+        $target = new ilNonEditableValueGUI(
+            $this->lng->txt('ecs_form_target_platform'),
+            'organisation'
+        );
+        $target->setValue($this->lookupOrganization());
+        $form->addItem($target);
+
+        // provider
+        $organisation = $this->getOrganisation();
+        if ($organisation instanceof ilECSOrganisation) {
+            $provider = new ilNonEditableValueGUI(
+                $this->lng->txt('organization'),
+                'provider'
+            );
+            $provider->setValue($organisation->getName());
+            $form->addItem($provider);
+        }
+
+        $consent = new ilCheckboxInputGUI(
+            $this->lng->txt('ecs_form_consent'), 'consent'
+        );
+        $consent->setValue("1");
+        $consent->setChecked($this->consents->hasConsented($this->mid));
+        $consent->setRequired(true);
+        $form->addItem($consent);
+
+        $user_data_fields = [];
+        foreach (['login',
+                  'firstname',
+                  'lastname',
+                  'email',
+                  'institution'
+                 ] as $field) {
+            $user_data_fields[] = $this->lng->txt('ecs_' . $field);
+        }
+        $listing = $this->ui_factory->listing()->unordered($user_data_fields);
+        $listing_html = $this->ui_renderer->render([$listing]);
+        $consent->setOptionTitle(
+            $this->lng->txt(
+                'ecs_form_consent_option_title'
+            ) . '<br />' . $listing_html
+        );
+        $submit = new ilHiddenInputGUI('cmd');
+        $submit->setValue('submit');
+        $form->addItem($submit);
+        return $form;
+    }
+
+    protected function getOrganisation() : ?ilECSOrganisation
+    {
+        $server_id = $this->importManager->lookupServerId($this->obj_id);
+        $community_reader = ilECSCommunityReader::getInstanceByServerId(
+            $server_id
+        );
+        try {
+            $part = $community_reader->getParticipantByMID($this->mid);
+            if ($part instanceof ilECSParticipant) {
+                return $part->getOrganisation();
+            }
+            return null;
+        } catch (ilECSConnectorException $e) {
+            return null;
+        }
+    }
+
+    protected function getGUIClassName() : string
+    {
+        return get_class($this->remote_gui);
+    }
+
+    /**
+     * @throws ilDatabaseException
+     * @throws ilObjectNotFoundException
+     */
+    protected function initRemoteObject() : ilRemoteObjectBase
+    {
+        $remote = ilObjectFactory::getInstanceByRefId($this->ref_id);
+        if (!$remote instanceof ilRemoteObjectBase) {
+            throw new ilObjectNotFoundException(
+                'Invalid ref_id given: ' . $this->ref_id
+            );
+        }
+        return $remote;
+    }
+}
```

Comment:
This is a coding standard violation.

I strongly recommend using `CaptainHook` or a respective inspection profile in PhpStorm to detect and fix such issues.